### ISSUE TITLE
Optimized image policy -- legacy-image-format -- which formats are modern formats? 

### ISSUE THREAD

**Issue by loonybear** [WICG/feature-policy#195]
#159 is proposing to ship a set of policies, including "[legacy-image-format](https://github.com/loonybear/feature-policy/blob/images/policies/optimized-images.md#legacy-image-formats)". 
Currently the default list of modern image formats is: JPEG, PNG, GIF, WEBP, and SVG

Did I miss anything. Are people happy with shipping this policy with the default list specified above?

**Comment by loonybear** [comment]
@clelland @dbaron @travisleithead @hober @jpchase

**Comment by eeeps** [comment]
JPEG-XR (Edge) and JPEG2000 (iOS) might belong on that list?

**Comment by niutech** [comment]
Please add [Animated PNG (APNG)](https://en.wikipedia.org/wiki/APNG) format and remove Animated GIF from the modern image formats list. APNG s supported by [majority](https://caniuse.com/#search=apng) of user agents, whereas GIF is bloated, inefficient, ancient and supports up to 256 colors.

**Comment by annevk** [comment]
I don't understand why we'd still require browsers to render the image. Couldn't it be advantageous that the browser knows it doesn't have to use certain decoders for a given context?

**Comment by ojanvafai** [comment]
Hmm...with the other image policies (like the image compression one) we need to download the image to know if it violates the policy. So we want to render it incorrectly so the developer fixes the issue rather than not noticing that the image didn't render.

For this policy, I don't recall if we had similar reasoning. It seems like we should be able to know up front that the image is the wrong format and not even initiate the request. @loonybear do you recall if there's something I'm missing here?

**Comment by domenic** [comment]
Note that we still need to download the image, or at least the first X bytes, in order to [sniff what type it is](https://mimesniff.spec.whatwg.org/#matching-an-image-type-pattern).

**Comment by annevk** [comment]
@ojanvafai but couldn't developers then simply apply a filter of sorts and ignore the policy? If you need something rendered, just make it render as opaque black then (though I guess you still need to support the formats in that case if you want to extract width/height, so perhaps it should also be a fixed size in that case).

**Comment by clelland** [comment]
Yes, there's no way to tell before hitting the network what format the image is; the URL is not any kind of indicator.

I can imagine that there might be architectures that could save memory or respond faster if they didn't have to load image codecs which weren't required. In that case, we'd want to spec this such that the image wasn't rendered at all. (And ignore the image dimensions, as though the image data was undecipherable. Practically, I would this policy to be used in conjunction with the 'unsized-media' policy anyway, so that some size would need to be set in markup or CSS.)

Also, we might need to go further than this to ensure that the codecs aren't accessible at all -- if something like `canvas.toDataURL` might also require them. Are there other APIs we'd need to restrict  as well?

**Comment by niutech** [comment]
Isn't "Content-Type: image/*" response header reliable?

**Comment by clelland** [comment]
@niutech, that still requires a request/response from the server (even if it's just a HEAD request) -- and still, that's not always sufficient. (@annevk's comment [here](https://github.com/WICG/feature-policy/issues/159#issuecomment-390880078) as well) A long history of incorrectly-configured web servers in the wild means that browsers pretty much have to do content-sniffing as well.

**Comment by ojanvafai** [comment]
@annevk I think this question is off topic for this issue of which image formats to consider legacy. Let's continue the conversation in https://github.com/WICG/feature-policy/issues/197.

**Comment by ayumihamsaki** [comment]
The line in the Spec

> The default list of modern image formats is: JPEG, PNG, GIF, WEBP, and SVG.

Some people could argue that and say the list of modern image formats on the internet is actually:

- png
- jpg
- jpeg
- svg
- webp
- gif
- apng
- mng
- flip
- bpg
- heif
- heic
- jxr
- jp2

My point is that I think you should re-word that line in the Spec. to say Feature Policy only covers the following image formats.
